contracts/contracts/EventManager.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+contract EventManager {
+    struct Event {
+        uint256 eventId;
+        uint16 eventType;
+        bytes data;
+    }
+
+    mapping(uint => Event[]) private _eventsByType;
+    uint private _eventBufferSize;
+    address public writer;
```

Comment:
do you think we need to store an owner in this contract? why not in the "main" contract instantiated on top of it, eg in `RandomnessGenerator.sol`? if you are to declare an owner here, then we'll need to implement an ownership transfer logic here as well. i'd make this contract as a "base" contact (or sort of an interface) and make operations with the owner isolated.